File: Javascript/4_function_4/forEarch.js
arr = [1, 2, 3, 4, 5];
sum = 0;
// arr.forEach(function(value,index,array) {
//     console.log("Value: " + value + " Index: " + index + " Array: " + array);
// });

arr.forEach(function(value) {
    sum += value;
});
console.log(sum);

File: Python practice/12 function/forEarch.py
list1 = [1, 2, 3, 4, 5]

# def forEach(list1,callback):
#     for i in list1:
#         callback(i, list1)

# def printValue(value,array):
#     print(f"Value: {value} Array: {array}")

def forEach(arr,callback):
    for i in range(len(arr)):
        callback(arr[i], i, arr)
        

def printValue(value,index,array):
    print(f"Value: {value} Index: {index} Array: {array}")

forEach(list1, printValue)

# forEach(list1, lambda value,index,array: print(f"Value: {value} Index: {index} Array: {array}"))


















